src/briefcase/commands/new.py

```diff
@@ -516,16 +645,89 @@ def verify_tools(self):
+            # safe to copy-paste the text, and that way also keep formatting and comments.
+            merged_pyproject = "\n\n\n".join((new_pyproject, pyproject))
+        else:
+            merged_pyproject = new_pyproject
+
+        with open(pyproject_file, "w") as file:
+            file.write(merged_pyproject)
+
     def __call__(
         self,
         template: Optional[str] = None,
         template_branch: Optional[str] = None,
+        target: Optional[Path] = None,
         **options,
     ):
         # Confirm host compatibility, and that all required tools are available.
         # There are no apps, so finalize() will be a no op on app configurations.
-        self.finalize()
 
-        return self.new_app(
-            template=template, template_branch=template_branch, **options
-        )
+        self.finalize()
+        if not target:
+            return self.new_app(
+                template=template, template_branch=template_branch, **options
+            )
+        elif not target.parent.exists():
+            raise ValueError("The parent directory of ``--target`` must exist.")
+        elif (target / "pyproject.toml").exists():
+            with open(target / "pyproject.toml", "rb") as file:
+                pyproject = tomllib.load(file)
+            if "tool" in pyproject and "briefcase" in pyproject["tool"]:
+                raise ValueError(
+                    f"[tool.briefcase] already in {target / 'pyproject.toml'}."
+                    " Cannot initialise briefcase for an application that already is packaged with briefcase."
+                )
+
+        with TemporaryDirectory() as output_path:
+            output_path = Path(output_path)
+            base_path = self.base_path
+            self.base_path = output_path
+            try:
+                app_name = self.new_app(
+                    template=template,
+                    template_branch=template_branch,
+                    target=target,
+                    **options,
+                )
+            finally:
+                self.base_path = base_path
+            project_dir = output_path / app_name
+
+            if target.exists():
+                self.merge_or_copy_pyproject(project_dir, target)
+                license_file = target / "LICENSE"
+                if not license_file.is_file():
+                    self.logger.warning(
+                        f"License file not found in {target}. You should either create a new license file in {target}"
+                        " or rename an already existing license file to LICENSE."
+                    )
+
+                changelog_file = target / "CHANGELOG"
+                if not changelog_file.is_file():
+                    self.logger.warning(
+                        f"Changelog file not found in {target}. You should either create a new changelog file in"
+                        f" {target} or rename an already existing changelog file to CHANGELOG."
+                    )
```

Comment:
There's one more resource that needs to be copied - the test entry script. This will be `tests/appname.py` in the generated template; that entry script should be copied into the `tests` folder.

Reply:
We're a bit unsure about what to do here. When looking closer at this, we found something we are unsure if is a problem. If we understand this correctly, we can specify a folder for the tests in pyproject.toml. But in [`tests/appname.py`](https://github.com/beeware/briefcase-template/blob/main/%7B%7B%20cookiecutter.app_name%20%7D%7D/tests/%7B%7B%20cookiecutter.module_name%20%7D%7D.py), it seems to be hardcoded (on [line 18](https://github.com/beeware/briefcase-template/blob/6e8a0d2777244d13b34a6572a5be45bd71a9a229/%7B%7B%20cookiecutter.app_name%20%7D%7D/tests/%7B%7B%20cookiecutter.module_name%20%7D%7D.py#L18)). I guess we could change the cookiecutter to use the provided test source directory (which we added to [our branch](https://github.com/yngvem/briefcase-template/blob/c2f67cb9ced448f6e6caf9d99ee9d3a7bf49a12e/%7B%7B%20cookiecutter.app_name%20%7D%7D/pyproject.toml#L21)) in [`tests/appname.py`](https://github.com/beeware/briefcase-template/blob/main/%7B%7B%20cookiecutter.app_name%20%7D%7D/tests/%7B%7B%20cookiecutter.module_name%20%7D%7D.py) as well, but we're unsure if hardcoding this path is the correct option. Could we somehow read this from the pyproject file or simply the name of the directory the tests lie in, or could this cause some issues?

Reply:
The way Briefcase runs tests is to run `python -m tests.appname` (or its equivalent) - so the `appname.py` is part of the briefcase contract; and it contains content that won't exist in any existing test harness (it needs to emit a termination sentinel). This is required because Briefcase isn't a literal Python interpreter, so you can't just "run `pytest` over a folder" - we need to set up the test suite inside the packaged app.

`pyproject.toml` specifies the additional source folders you want to copy into your app when it's running in test mode; the simplest approach is to put all your tests in a `tests` folder (and most projects follow this convention anyway, so it's not much of a burden). However, just having those tests isn't enough - the `tests.appname` module entry point needs to exist.